components/tidb_query/src/codec/mysql/duration.rs

```diff
@@ -1126,6 +1135,183 @@ mod tests {
+                false,
+            ),
+            (
+                8385959,
+                6,
+                Ok(Duration::parse("838:59:59".as_bytes(), 6).unwrap()),
+                false,
+            ),
+            (
+                -101010,
+                0,
+                Ok(Duration::parse("-10:10:10".as_bytes(), 0).unwrap()),
+                false,
+            ),
+            (
+                -101010,
+                5,
+                Ok(Duration::parse("-10:10:10".as_bytes(), 5).unwrap()),
+                false,
+            ),
+            (
+                -8385959,
+                0,
+                Ok(Duration::parse("-838:59:59".as_bytes(), 0).unwrap()),
+                false,
+            ),
+            (
+                -8385959,
+                6,
+                Ok(Duration::parse("-838:59:59".as_bytes(), 6).unwrap()),
+                false,
+            ),
+            // will overflow
+            (
+                8385960,
+                0,
+                Ok(Duration::parse("838:59:59".as_bytes(), 0).unwrap()),
+                true,
+            ),
+            (
+                8385960,
+                1,
+                Ok(Duration::parse("838:59:59".as_bytes(), 1).unwrap()),
+                true,
+            ),
+            (
+                8385960,
+                5,
+                Ok(Duration::parse("838:59:59".as_bytes(), 5).unwrap()),
+                true,
+            ),
+            (
+                8385960,
+                6,
+                Ok(Duration::parse("838:59:59".as_bytes(), 6).unwrap()),
+                true,
+            ),
+            (
+                -8385960,
+                0,
+                Ok(Duration::parse("-838:59:59".as_bytes(), 0).unwrap()),
+                true,
+            ),
+            (
+                -8385960,
+                1,
+                Ok(Duration::parse("-838:59:59".as_bytes(), 1).unwrap()),
+                true,
+            ),
+            (
+                -8385960,
+                5,
+                Ok(Duration::parse("-838:59:59".as_bytes(), 5).unwrap()),
+                true,
+            ),
+            (
+                -8385960,
+                6,
+                Ok(Duration::parse("-838:59:59".as_bytes(), 6).unwrap()),
+                true,
+            ),
+            // will truncated
+            (8376049, 0, Err(Error::truncated_wrong_val("", "")), false),
+            (8375960, 0, Err(Error::truncated_wrong_val("", "")), false),
+            (8376049, 0, Err(Error::truncated_wrong_val("", "")), false),
+            // TODO, add test for num>=10000000000
+            //  after Duration::from_f64 had impl logic for num>=10000000000
+            // (10000000000, 0, Ok(Duration::parse("0:0:0".as_bytes(), 0).unwrap())),
+            // (10000235959, 0, Ok(Duration::parse("23:59:59".as_bytes(), 0).unwrap())),
+            // (10000000000, 0, Ok(Duration::parse("0:0:0".as_bytes(), 0).unwrap())),
```

Comment:
I suggest add test case for num>=10000000000 here (assert the wrong behavior in TiKV).
In case, If someone implement this funciton or modify the logic, the test case will failed, and it will inform he/she to remove the old test case and use new test case.